go/service/chat_local.go

```diff
@@ -130,17 +210,76 @@ getinbox:
 		}
 	}
 
-	id, err := h.NewConversationLocal(ctx, chat1.ConversationIDTriple{
-		Tlfid:     tlfID,
-		TopicType: arg.TopicType,
-		// TopicID filled by server?
-	})
-	if err != nil {
-		return ids, err
+	return conversations, nil
+}
+
+// getConversationInfo locates the conversation by using id, and returns with
+// all fields filled in conversationInfo, along with a ConversationIDTriple
+//
+// TODO: cache
+func (h *chatLocalHandler) getConversationInfo(ctx context.Context, id chat1.ConversationID) (conversationInfo keybase1.ConversationInfoLocal, triple chat1.ConversationIDTriple, err error) {
+	ipagination := &chat1.Pagination{Num: 20}
+getinbox:
+	for i := 0; i < 10000; /* in case we have a server bug */ i++ {
+		iview, err := h.GetInboxLocal(ctx, ipagination)
+		if err != nil {
+			return conversationInfo, triple, err
+		}
+
+		for _, conversation := range iview.Conversations {
+			if id == conversation.Metadata.ConversationID {
+				triple = conversation.Metadata.IdTriple
+				conversationInfo.Id = id
+				conversationInfo.TopicType = triple.TopicType
+
+				if len(conversation.MaxHeaders) == 0 {
+					return conversationInfo, triple, errors.New("empty conversation found")
```

Comment:
When starting a new conversation, the client has a `ConversationIDTriple` derived from TlfName, and can use it to insert the first message. This message has TlfName as part of signed header, which can be relied on by subsequent messages.

As to security concerns, I was convinced by @strib yesterday. This is part of enforcing using a TlfName as part of identifying a conversation, since it involves running identify. If we merely use ConversationIDTriple'a TlfId, we would need a kbfs rpc for translating TlfId->TlfName. If we make a mistake in the future by indexing (caching) keys with TlfID, it might open up possibilities for a malicious server to send wrong TlfId and confuse the client to encrypt for wrong conversation